src/storage.js

```diff
@@ -0,0 +1,430 @@
+ * downloaded from eff.org. The keys are the domains which should be blocked.
+ * The values are simply 'true'
+ *
+ * { 
+ *   "maps.google.com": true,
+ *   "creativecommons.org": true,
+ * }
+ **/
+
+// Cache of BadgerStorage objects
+var badgerPen = {};
+
+
+var initialize = function(callback){
+  console.log('loading badgers into the pen');
+  var storage_objects = [
+    "snitch_map",
+    "action_map",
+    "cookieblock_list",
+    "supercookie_domains",
+    "dnt_hashes",
+    "settings_map"
+  ];
+
+  _initializeCache(storage_objects, callback);
+};
+
+var getScore = function(action){
+  switch(action){
+    case pb.NO_TRACKING: return 0;
+    case pb.ALLOW: return 1;
+    case pb.BLOCK: return 2;
+    case pb.COOKIEBLOCK: return 3;
+    case pb.DNT: return 4;
+    default: return 5; 
+  }
+};
+
+/**
+ * get the current presumed action for a specific fqdn, ignoring any rules for subdomains
+ * below or above it
+ * @param {Object} domain domain object from action_map
+ * @returns {String} the presumed action for this FQDN
+ **/
+var getActionForFqdn = function(domain){
+  if(domain.userAction){ return domain.userAction; }
+  if(domain.dnt){ return pb.DNT; } 
+  if(domain.heuristicAction){ return domain.heuristicAction; } 
+  return pb.NO_TRACKING;
+};
+
+/**
+ * adds a heuristic action for a domain
+ * @param {String} domain Domain to add
+ * @param {String} action The heuristic action to take
+ */
+var setupHeuristicAction = function(domain, action){
+  _setupDomainAction(domain, action, "heuristicAction");
+};
+
+/**
+ * Sets up a domain for DNT
+ * @param {String} domain Domain to add
+ */
+var setupDNT = function(domain){
+  _setupDomainAction(domain, "true", "dnt"); 
+};
+  
+/**
+* remove DNT setting from a domain
+* @param domain FQDN string
+**/
+var revertDNT = function(domain){
+  _setupDomainAction(domain, null, "dnt");
+};
+
+var touchDNTRecheckTime = function(domain, time){
+  var action_map = getBadgerStorageObject('action_map');
+  var domainObj = action_map.getItem(domain);
+  domainObj.nextUpdateTime = time;
+  action_map.setItem(domain, domainObj);
+};
+
+var getNextUpdateForDomain = function(domain){
+  var action_map = getBadgerStorageObject('action_map');
+  if(action_map.hasItem(domain)){
+    return action_map.getItem(domain).nextUpdateTime;
+  } else {
+    return 0;
+  }
+};
+
+/**
+ * update DNT policy hashes
+ */
+var updateDNTHashes = function(hashes){
+  var dnt_hashes = getBadgerStorageObject('dnt_hashes');
+  dnt_hashes.updateObject(_.invert(hashes));
+};
+
+/**
+ * adds a heuristic action for a domain
+ * @param {String} domain Domain to add
+ * @param {String} action The heuristic action to take
+ */
+var setupUserAction = function(domain, action){
+  _setupDomainAction(domain, action, "userAction");
+};
+  
+/**
+* remove user set action from a domain
+* @param domain FQDN string
+**/
+var revertUserAction = function(domain){
+  _setupDomainAction(domain, null, "userAction");
```

Comment:
Same as my comment above about reverting the DNT setting. Maybe either this should be updated to `_setupDomainAction(domain, "", "userAction");` or the [action_map comments](https://github.com/EFForg/privacybadgerchrome/blob/weapi/src/storage.js#L33) should be updated to show null for domains without a user override.